packages/wrangler/src/pages.tsx

```diff
@@ -1172,6 +1177,205 @@ export const pages: BuilderCallback<unknown, unknown> = (yargs) => {
           render(<Table data={data}></Table>);
         }
       )
+    )
+    .command(
+      "publish [directory]",
+      "Compile a folder of Cloudflare Pages Functions into a single Worker",
+      (yargs) => {
+        return yargs
+          .positional("directory", {
+            type: "string",
+            default: "functions",
+            description: "The directory of Pages Functions",
+          })
+          .options({
+            project: {
+              type: "string",
+              // default: "_worker.js",
+              description:
+                "The name of the project you want to list deployments for",
+            },
+            branch: {
+              type: "string",
+              // default: "_worker.js",
+              description:
+                "The branch of the project you want to list deployments for",
+            },
+            "commit-hash": {
+              type: "string",
+              // default: "_worker.js",
+              description:
+                "The branch of the project you want to list deployments for",
+            },
+            "commit-message": {
+              type: "string",
+              // default: "_worker.js",
+              description:
+                "The branch of the project you want to list deployments for",
+            },
+            "commit-dirty": {
+              type: "boolean",
+              default: false,
+              description:
+                "The branch of the project you want to list deployments for",
+            },
+          });
+      },
+      async (args) => {
+        const {
+          directory,
+          project,
+          // branch,
+          // commitHash,
+          // commitMessage,
+          // commitDirty,
+        } = args;
+
+        // TODO: Add check for dirty git state if commitDirty is not true
+
+        const config = readConfig(args.config as ConfigPath, args);
+        const accountId = await requireAuth(config);
+
+        type File = {
+          content: Buffer;
+          metadata: Metadata;
+        };
+
+        type Metadata = {
+          sizeInBytes: number;
+          hash: string;
+        };
+
+        const IGNORE_LIST = [
+          "_worker.js",
```

Comment:
@GregBrimble Any files I'm missing that should be in this?

Reply:
We should ignore symlinks and `node_modules` as well